### ISSUE TITLE
The issue of multiplicative noise addition

### ISSUE THREAD

**Issue by BHC1205** [hustvl/DiffusionDriveV2#17]
In the paper, by "multiplicative noise," I understand it to mean applying independent Gaussian noise in the random x and y directions, and then multiplying this Gaussian noise with the trajectory—is that correct?

However, the problem lies in the fact that, in the code, when computing the policy gradient, a Gaussian probability density is used. Yet, this noise-addition method does not actually follow Gaussian sampling—although the weighting terms are obtained via Gaussian sampling, the process of applying the noise to the trajectory itself does not conform to Gaussian sampling.

Therefore, I would like to ask whether this presents a logical inconsistency, and whether there are any plans for subsequent improvements. 

Looking forward to your reply.

**Comment by Doctor-James** [comment]
This part isn't mathematically perfect, so you can treat it more like an engineering approximation. From an ML standpoint, you can basically look at this as an MSE loss where the constant terms drop out during backprop. The physical meaning here is that if $adv > 0$, it acts as a 'pull' loss that draws the generated trajectory closer to the exploration trajectory; if $adv < 0$, it becomes a 'push' loss.

```
log_prob = (
    -((prev_sample.detach() - prev_sample_mean) ** 2) / (2 * (std_dev_t_mul**2))
    - torch.log(std_dev_t_mul)
    - torch.log(torch.sqrt(2 * torch.as_tensor(math.pi)))
)   
````
In fact, we attempted a more mathematically rigorous version, but it proved to be unstable during training. Consequently, we decided to retain the original algorithm.
```
var = prev_sample_mean.pow(2) * std_dev_t_mul.pow(2) + std_dev_t_add.pow(2)
var = torch.clamp(var, min=1e-10)
delta = prev_sample.detach() - prev_sample_mean
log_prob = -0.5 * delta.pow(2) / var -0.5 * torch.log(2 * math.pi * var)
```

**Comment by BHC1205** [comment]
I suspect that in certain static scenarios, the var might have hit the boundary value of 0, leading to abnormal training data. Was any special handling applied to this case during training?

**Comment by Doctor-James** [comment]
From an ML perspective, I think the original formula is sufficient and offers stable training, so we didn't investigate it further.

**Comment by BHC1205** [comment]
Thank you for your reply.
